src/main/java/de/tum/www1/orion/ui/browser/BrowserWebView.kt

```diff
@@ -0,0 +1,82 @@
+package de.tum.www1.orion.ui.browser
+
+import com.intellij.openapi.components.ServiceManager
+import com.intellij.openapi.components.service
+import com.intellij.openapi.project.Project
+import com.intellij.ui.jcef.JBCefApp
+import com.intellij.ui.jcef.JBCefBrowser
+import com.intellij.ui.jcef.JBCefClient
+import com.intellij.ui.jcef.JBCefJSQuery
+import com.intellij.util.messages.Topic
+import de.tum.www1.orion.connector.ide.build.OrionBuildConnector
+import de.tum.www1.orion.connector.ide.exercise.OrionExerciseConnector
+import de.tum.www1.orion.connector.ide.shared.OrionSharedUtilConnector
+import de.tum.www1.orion.connector.ide.vcs.OrionVCSConnector
+import de.tum.www1.orion.settings.OrionSettingsProvider
+import de.tum.www1.orion.ui.OrionRouter
+import de.tum.www1.orion.util.getPrivateProperty
+import org.cef.CefApp
+import org.cef.CefSettings
+import org.cef.browser.CefBrowser
+import org.cef.browser.CefFrame
+import org.cef.handler.CefLoadHandlerAdapter
+import java.util.*
+
+class BrowserWebView(val project: Project) {
+    val jbCefBrowser: JBCefBrowser
+    val jsQuery: JBCefJSQuery
+    val client: JBCefClient
+
+    /**
+     * Inits the actual browser panel. We use a JFXPanel in a [WebView] gets initialized. This web view only
+     * displays the ArTEMiS Angular webapp containing a few adaptions, so that we only show the most important information
+     * in the IDE.
+     */
+    init {
+        //Dumb check but must be called to perform initialization
+        if (!JBCefApp.isSupported()) {
+            // Fallback to an alternative browser-less solution
```

Comment:
Please show the user a warning/error, in this case until we have a fallback solution implemented